### DISCUSSION TITLE
Solution for Supabase storage: unusually slow when fetching and displaying image?

### DISCUSSION THREAD

**Discussion by JasonChiu-dev** [supabase/supabase#11605] [Questions] [answered]
Hello everyone!

I have been using supabase for a while, I like it very much and I am very satisfied with the service of supasbse.

Recently, I tried to use the supasbe storage bucket (**_private_** bucket) to solve my client's new needs - storing work photos and recalling that photo...etc. This is the first time I have used supabase storage.

When I call photos stored in the supabase storage bucket (
using

**signedUrl = await supabase.storage.from(bucket).createSignedUrl(path, 60);**

to get signedUrl and use the Image.network widget to display the picture

**Image.network(signedUrl, fit: BoxFit.contain);**

), and found that the response speed is extremely slow, even if I only call one photo. This does not match the original response speed of my APP, and it becomes lagged when displaying pictures.

**I wonder if you have encountered the same problem? Looking forward to your assistance.**

By the way, 

I found if I set the **private** bucket to **public** and replace

signedUrl = await supabase.storage.from(bucket).createSignedUrl(path, 60);
Image.network(signedUrl, fit: BoxFit.contain);

by

publicUrl = supabase.storage.from(bucket).getPublicUrl(path);
Image.network(publicUrl, fit: BoxFit.contain);

It speeds up.

**The point is that I don't want to make the bucket public, but want to maintain the original private bucket status.**

**Is there any way by using createSignedUrl and speed up displaying the image?**

**Or a workaround provided by Supabase?**

**Comment by JasonChiu-dev** [comment, marked as answer]
After actual testing, it was finally found that the use of

```
downloadedFile = await supabase.storage.from(bucket).download(path); 

Image.memory(downloaedFile);
```

**In this way, both the speed of image display and the use of private storage bucket can be maintained.**

Share it with you, I hope it will be helpful.

**Comment by devidevio** [comment]
As per Storage documentation, public buckets are always faster than private buckets, see here:
* https://supabase.com/docs/guides/storage/access-control#public-and-private-buckets
* https://supabase.com/docs/guides/storage/cdn#public-vs-private-buckets

> [...]For public buckets, you can access the assets directly without a token or an Authorisation header.[...]

> Objects in public buckets do not require any Authorization to access objects. This leads to a better cache hit rate compared to private buckets. For private buckets, permissions for accessing each object is checked on a per user level. For example, if two different users access the same object in a private bucket from the same region, it results in a cache miss for both the users since they might have different security policies attached to them. On the other hand, if two different users access the same object in a public bucket from the same region, it results in a cache hit for the second user.
